rules/TypeDeclaration/Rector/Class_/ReturnTypeFromStrictTernaryRector.php

```diff
@@ -0,0 +1,186 @@
+use PHPStan\Type\ConstantType;
+use PHPStan\Type\GeneralizePrecision;
+use PHPStan\Type\Type;
+use Rector\Core\Rector\AbstractRector;
+use Rector\Core\ValueObject\PhpVersionFeature;
+use Rector\PHPStanStaticTypeMapper\Enum\TypeKind;
+use Rector\TypeDeclaration\ValueObject\TernaryIfElseTypes;
+use Rector\VendorLocker\ParentClassMethodTypeOverrideGuard;
+use Rector\VersionBonding\Contract\MinPhpVersionInterface;
+use Symplify\RuleDocGenerator\ValueObject\CodeSample\CodeSample;
+use Symplify\RuleDocGenerator\ValueObject\RuleDefinition;
+
+/**
+ * @see \Rector\Tests\TypeDeclaration\Rector\Class_\ReturnTypeFromStrictTernaryRector\ReturnTypeFromStrictTernaryRectorTest
+ */
+final class ReturnTypeFromStrictTernaryRector extends AbstractRector implements MinPhpVersionInterface
+{
+    public function __construct(
+        private readonly ParentClassMethodTypeOverrideGuard $parentClassMethodTypeOverrideGuard
+    ) {
+    }
+
+    public function getRuleDefinition(): RuleDefinition
+    {
+        return new RuleDefinition('Add method return type based on strict ternary values', [
+            new CodeSample(
+                <<<'CODE_SAMPLE'
+final class SomeClass
+{
+    public function getValue($number)
+    {
+        return $number ? 100 : 500;
+    }
+}
+CODE_SAMPLE
+
+                ,
+                <<<'CODE_SAMPLE'
+final class SomeClass
+{
+    public function getValue($number): int
+    {
+        return $number ? 100 : 500;
+    }
+}
+CODE_SAMPLE
+            ),
+        ]);
+    }
+
+    /**
+     * @return array<class-string<Node>>
+     */
+    public function getNodeTypes(): array
+    {
+        return [Class_::class];
+    }
+
+    /**
+     * @param Class_ $node
+     */
+    public function refactor(Node $node): ?Node
+    {
+        $hasChanged = false;
+
+        foreach ($node->getMethods() as $classMethod) {
+            if ($classMethod->returnType instanceof Node) {
+                continue;
+            }
+
+            $onlyStmt = $classMethod->stmts[0] ?? null;
+            if (! $onlyStmt instanceof Return_) {
+                continue;
+            }
+
+            if (! $onlyStmt->expr instanceof Ternary) {
+                continue;
+            }
+
+            $ternary = $onlyStmt->expr;
+
+            // has scalar in if/else of ternary
+            $ternaryIfElseTypes = $this->matchScalarTernaryIfElseTypes($ternary);
+            if (! $ternaryIfElseTypes instanceof TernaryIfElseTypes) {
+                continue;
+            }
+
+            $ifType = $ternaryIfElseTypes->getFirstType();
+            $elseType = $ternaryIfElseTypes->getSecondType();
+
+            if (! $this->areTypesEqual($ifType, $elseType)) {
+                continue;
+            }
+
+            $returnTypeNode = $this->staticTypeMapper->mapPHPStanTypeToPhpParserNode($ifType, TypeKind::RETURN);
+
+            if (! $this->parentClassMethodTypeOverrideGuard->shouldSkipReturnTypeChange($classMethod, $ifType)) {
```

Comment:
shoud skip should remove `! ` part then:

```suggestion
         if ($this->parentClassMethodTypeOverrideGuard->shouldSkipReturnTypeChange($classMethod, $ifType)) {
```

as `continue` instead of processing.

Reply:
Cool :+1: